jfr-daemon/src/test/java/com/newrelic/jfr/daemon/JFRUploaderTest.java

```diff
@@ -90,23 +87,51 @@ void testBufferingThrowsExceptionIsHandled() throws Exception {
         .when(recordedEventBuffer)
         .bufferEvents(filePath, recordingFile);
 
-    var testClass =
-        new JFRUploader(
-            telemetryClient, recordedEventBuffer, eventConverter, x -> recordingFile, deleter);
+    JFRUploader testClass = buildTestClass();
 
     testClass.handleFile(filePath);
     // no exception, but we still try and send
     verify(telemetryClient).sendBatch(expectedMetricBatch);
     verify(telemetryClient).sendBatch(expectedEventBatch);
   }
 
+  @Test
+  void testSkipsIfNotReady() {
+    JFRUploader testClass = buildTestClass(false);
+    testClass.handleFile(filePath);
+    verifyNoInteractions(eventConverter);
+    verifyNoInteractions(telemetryClient);
+  }
+
+  private JFRUploader buildTestClass() {
+    return buildTestClass(true);
+  }
+
+  private JFRUploader buildTestClass(boolean ready) {
+    var testClass =
+        JFRUploader.builder()
+            .telemetryClient(telemetryClient)
+            .recordedEventBuffer(recordedEventBuffer)
+            .eventConverter(eventConverter)
+            .recordingFileOpener(x -> recordingFile)
+            .fileDeleter(deleter)
+            .readinessCheck(new AtomicBoolean(ready))
+            .build();
+    return testClass;
+  }
+
   @Test
   public void testConvertThrowsExceptionIsHandled() throws Exception {
     doThrow(new RuntimeException("kaboom!")).when(eventConverter).convert(recordedEventBuffer);
 
     var testClass =
-        new JFRUploader(
-            telemetryClient, recordedEventBuffer, eventConverter, x -> recordingFile, deleter);
+        JFRUploader.builder()
```

Comment:
AFAICT, using `buildTestClass` would work here?  I see this builder is missing the readiness check but that doesn't seem to matter for this convert test.

Reply:
Ooh, good catch.  I'll change that.